internal/provider/provider.go

```diff
@@ -311,12 +319,21 @@ func (p *provider) HandleMountRequest(ctx context.Context, cfg config.Config, fl
 	}, nil
 }
 
-func generateObjectVersion(secret config.Secret, content []byte) (*pb.ObjectVersion, error) {
-	hash := sha256.New()
+func generateObjectVersion(secret config.Secret, hmacKey []byte, content []byte) (*pb.ObjectVersion, error) {
+	// If something went wrong with generating the HMAC key, we log the error and
+	// treat generating the version as best-effort instead, as delivering the secret
+	// is generally more critical to workloads than assigning a version for it.
+	if hmacKey == nil {
+		return &pb.ObjectVersion{
+			Id:      secret.ObjectName,
+			Version: "",
+		}, nil
+	}
+
 	// We include the secret config in the hash input to avoid leaking information
 	// about different secrets that could have the same content.
-	_, err := hash.Write([]byte(fmt.Sprintf("%v:%s", secret, content)))
-	if err != nil {
+	hash := hmac.New(sha256.New, hmacKey)
+	if _, err := hash.Write([]byte(fmt.Sprintf("%v:%s", secret, content))); err != nil {
```

Comment:
While we're here changing the hash, I think it might be prudent to use something with other than `%v` to serialize the secret, like JSON or gob or `encoding/binary`, so that it won't contain, say, memory addresses and has more guarantees about consistent encoding.